Lesson 7 part 2

mwrobert opened the pull request:
Zaimplementowałem konfiguracje poprzez pliki konfiguracyjne (*.properties).

JoannaBiegala committed:
part 1

JoannaBiegala committed:
part 1

JoannaBiegala committed:
part 1

JoannaBiegala committed:
part 1

JoannaBiegala committed:
part 1

JoannaBiegala committed:
part 1

JoannaBiegala committed:
part 1

mwrobert committed:
Project configuration and logging

JoannaBiegala commented on build.gradle:
```diff
@@ -1,53 +1,70 @@
+
+    implementation "com.fasterxml.jackson.core:jackson-databind:$jacksonVersion"
+    implementation "com.fasterxml.jackson.datatype:jackson-datatype-jsr310:$jacksonVersion"
+    implementation "com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:$jacksonVersion"
```
Te powyższe 3 zależności możesz usunąć ponieważ zawierają się w org.springframework.boot:spring-boot-starter-web.
Można to zobaczyć uruchamiając w terminalu ./gradlew dependencies

KacperKropidlowski commented on build.gradle:
```diff
@@ -1,31 +1,24 @@
-        lombokVersion = '1.18.24'
         jacksonVersion = '2.11.1'
+        lombokVersion = '1.18.24'
         spockVersion = '2.1-groovy-3.0'
```
Można podbić wersje spocka do 2.2-groovy-3.0

mwrobert replied:
Zrobione

KacperKropidlowski commented on build.gradle:
```diff
@@ -34,24 +27,44 @@ dependencies {
 
     compileOnly "org.projectlombok:lombok:${lombokVersion}"
 
+    implementation "com.fasterxml.jackson.core:jackson-databind:$jacksonVersion"
+    implementation "com.fasterxml.jackson.datatype:jackson-datatype-jsr310:$jacksonVersion"
+    implementation "com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:$jacksonVersion"
```
Spring zawiera Jackson'a. Możemy usunąć go z dependencies, ponieważ jest transient.

mwrobert replied:
Usunięte

KacperKropidlowski commented on src/main/java/pl/futurecollars/invoice/db/file/FileDatabaseConfiguration.java:
```diff
@@ -2,39 +2,66 @@
     return new FileRepository(databasePath, filesService, jsonService, idService);
   }
 
+  @Bean
+  @ConditionalOnProperty(value = "database.type", havingValue = "in-memory")
+  public Database memoryRepository() {
+
+    System.out.println("Running on memoryRepository");
+
+    return new MemoryRepository();
+  }
+
+  @Bean
+  public InvoiceService invoiceService(Database database) {
+    return new InvoiceService(database);
+  }
+
```
Ta klasa jest w folderze plikowej bazy danych, i zawiera konfigurację plikowej bazy danych. Wydaje mi się, że powinniśmy wynieść konfigurację do folderu db i zmienić nazwę na DatabseConfiguration, albo zrobić klasę MemoryDatabaseConfiguration i przenieść ją do db/memory i tam umieścić te linijki.

mwrobert replied:
Zmieniłem nazwę na DatabseConfiguration i umieściłem w paczce pl.futurecollars.invoice.db

KacperKropidlowski commented on src/main/resources/application-memory.properties:
```diff
@@ -0,0 +1,2 @@
+server.port=8081
+database.type=in-memory
```
Dlaczego nie użyć nazw profili dev, prod i test?

mwrobert replied:
zmieniłęm nazwy profili

KacperKropidlowski reviewed (commented):
Kilka pytań i uwag.

mwrobert committed:
Project configuration and logging update:
- build.grile:
  - changed spock version to '2.2-groovy-3.0'
  - removed Jackson from dependencies
- moved configuration do DatabaseConfiguration class
- changed profiles names

KacperKropidlowski commented on src/main/resources/application-test-file.properties:
```diff
@@ -0,0 +1,4 @@
+server.port=8081
+database.type=in-file
+database.path=test_db/invoices.json
+database.idpath=test_db/nextId.txt
```
Po obejrzeniu wykładu zauważyłem, że tu jednak jest błąd. Patryk mówił, że propertiesy testowe powinny się znajdować w resources w katalogu test. Jak włączysz wykład to jest o tym mowa 1:28:40.

mwrobert replied:
Zrobiłem kopię application.properties w src/test/resources folder i przesunąłem tam application-test.properties.
Nowy PR do sprawdzenia.

JoannaBiegala commented:
Sprawdziłam, wg mnie jest dobrze.

patryk-stepniak commented on src/main/java/pl/futurecollars/invoice/db/DatabaseConfiguration.java:
```diff
@@ -0,0 +1,67 @@
+@Configuration
+public class DatabaseConfiguration {
+
+  @Bean
```
fileService oraz jsonService są potrzebne tylko w przypadku plikowej bazy danych

mwrobert replied:
Dałem teraz warunek dla tych 2 bean'ów tylko dla plikowej bazy danych

patryk-stepniak replied:
Z tego co widzę w kodzie to warunków brak, za to pojawiły się adnotacje @Service w obu tych klasach.

patryk-stepniak commented on src/main/resources/application-test.properties:
```diff
@@ -0,0 +1,4 @@
+server.port=8081
```
Te propertiesy są do testów? Jeśli tak to nie powinny znajdować się w src.

mwrobert replied:
Mamy 3 configuracje:
- prod (file or memory)
- dev (file or memory)
- test (file or memory)
Przeniosłem teraz test to test/resources folder

patryk-stepniak commented on src/test/groovy/pl/futurecollars/invoice/controller/InvoiceControllerUnitTest.groovy:
```diff
@@ -190,11 +190,14 @@ class InvoiceControllerUnitTest extends Specification {
     }
 
     private void resetFile(String filePath) {
+
```
coś się rozjechało

mwrobert replied:
Usunąłem puste linie

patryk-stepniak reviewed (changes requested):
Kilka uwag

mwrobert committed:
Update:
- moved test configuration files to test/resources folder
- updated DatabaseConfiguration to configure FileService and JsonService beans only for in-file configuration
- created BeanTestConfiguration to configure JsonService bean for tests
- removed empty lines from InvoiceControllerUnitTest resetFile helper method

mwrobert committed:
Merge branch 'master' into Lesson_7_part_2

KacperKropidlowski committed:
Corrections + logger

patryk-stepniak reviewed (changes requested):
Cały czas FilesService oraz JsonService są tworzone zawsze, niezależnie od wybranego profilu, a przecież potrzebujemy ich tylko w przypadku plikowej bazy danych.

mwrobert committed:
Updates:
- DatabaseConfiguration: set @ConditionalOnProperty for FileService and JsonService beans to be created only for in-file database

patryk-stepniak commented:
Jest ok, ale są konflikty. Zostawiam approve'a, zmergujcie sami po rozwiązaniu konfliktów.

KacperKropidlowski committed:
Merge branch 'master' into Lesson_7_part_2

KacperKropidlowski committed:
Correct version
